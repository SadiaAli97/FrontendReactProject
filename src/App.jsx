import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { ToggleOptions } from './components/ToggleOptions';
import { TableView } from './containers/TableView';
import { getInstalledSdks, getUninstalledSdks } from './utils/data';
import { loadSdkData, setSelectedToggleOption } from './actions/sdks';
import { ISDKS } from './constants/app-constants';
import './css/sdksview.css';

const App = ({selectedToggleOption, loadSdkData, setSelectedToggleOption, installedSdks, uninstalledSdks}) => {
  
  useEffect(() => {
    loadSdkData(getInstalledSdks(), getUninstalledSdks());
    setSelectedToggleOption(ISDKS);
  }, []);

  const handleChange = (newAlignment) => {
    setSelectedToggleOption(newAlignment);
  };


  return (
    <div className="App">
      <div className="toggle-options">
        <ToggleOptions
          selectedToggle={selectedToggleOption}
          toggleChange={handleChange}
        />
      </div>
      <TableView
        data={selectedToggleOption ? (selectedToggleOption === ISDKS ? installedSdks : uninstalledSdks) : []}
        selectedToggleOption={selectedToggleOption}
      />
    </div>
  );
}

const mapStateToProps = (state) => ({
  selectedToggleOption: state.sdks.selectedToggleOption,
  installedSdks: state.sdks.installedSdks,
  uninstalledSdks: state.sdks.uninstalledSdks,
})

const mapDispatchToProps = (dispatch) => ({
  loadSdkData: (installedSdks, uninstalledSdks) => dispatch(loadSdkData(installedSdks, uninstalledSdks)),
  setSelectedToggleOption: (option) => dispatch(setSelectedToggleOption(option)),
})

export default connect(mapStateToProps, mapDispatchToProps)(App);