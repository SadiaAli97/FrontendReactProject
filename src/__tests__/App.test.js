import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from '../App';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store' 

const middlewares = []
const mockStore = configureStore(middlewares)

const initialState = {
  sdks: {
    selectedToggleOption: 'isdks',
    installedSdks: [
      {
        "id": "1",
        "name": "SDK 1",
        "categories": ["Backend"],
        "firstSeenDate": "2017-11-16T10:48:53.000-08:00",
        "lastSeenDate": "2019-07-16T13:24:39.000-07:00",
        "__typename": "Sdk"
      },
    ],
    uninstalledSdks: [
    {
        "id": "1",
        "name": "Sdk 1",
        "categories": ["App Performance Management"],
        "firstSeenDate": "2017-11-16T10:48:53.000-08:00",
        "lastSeenDate": "2019-03-15T06:12:41.000-07:00",
        "__typename": "Sdk"
      },
    ],
  }
}

const store = mockStore(initialState)

it('renders App', () => {
  const testRenderer = TestRenderer.create(<Provider store={store}>
      <App />
    </Provider>
  );
  expect(testRenderer.toJSON()).toMatchSnapshot()

});