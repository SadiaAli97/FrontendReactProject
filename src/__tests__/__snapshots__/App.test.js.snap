// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div
  className="App"
>
  <div
    className="toggle-options"
  >
    <div
      className="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
      role="group"
    >
      <button
        aria-pressed={true}
        className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="isdks"
      >
        Installed
      </button>
      <button
        aria-pressed={false}
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="uisdks"
      >
        Uninstalled
      </button>
    </div>
  </div>
  <div
    className="table-outline"
  >
    <div
      className="MuiBox-root css-i9gxme"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
        >
          <div
            className="heading"
          >
            Installed SDK's
          </div>
          <div
            className="subheading"
          >
            Latest Update:
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            className="listcount"
          >
            1
          </div>
        </div>
      </div>
    </div>
    <div
      className="divider"
    />
    <div
      className="MuiBox-root css-i9gxme"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            className="list-item"
          >
            <div
              className="category"
            >
              Backend
            </div>
            <div
              className="items"
            >
              <div
                className="item"
              >
                SDK 1 2019-07-16T13:24:39.000-07:00
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
